File: solutions/cpp/Palindrome Linked List.cpp
#include <stack>
struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    bool isPalindrome(ListNode *head)
    {
        std::stack<ListNode *> s;
        ListNode *dummy = head;
        while (dummy)
        {
            s.push(dummy);
            dummy = dummy->next;
        }
        while (head)
        {
            ListNode *node = s.top();
            if (node->val != head->val)
                return false;
            s.pop();
            head = head->next;
        }
        return true;
    }
};

File: solutions/Javascript/Palindrome Linked List.js
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}

var isPalindrome = function (head) {
  let res = "";
  let node = head;
  while (node) {
    res = res + node.val.toString();
    node = node.next;
  }
  let i = 0,
    j = res.length - 1;
  while (i < j) {
    if (res[i] !== res[j]) {
      return false;
    }
    i = i + 1;
    j = j - 1;
  }
  return true;
};
